modules/wallet/seed.go

```diff
@@ -42,6 +44,27 @@ func generateSpendableKey(seed modules.Seed, index uint64) spendableKey {
 	}
 }
 
+// generateKeys generates n keys from seed, starting from index start.
+func generateKeys(seed modules.Seed, start, n uint64) []spendableKey {
+	// generate in parallel, one goroutine per core.
+	keys := make([]spendableKey, n)
+	var wg sync.WaitGroup
+	wg.Add(runtime.NumCPU())
+	for cpu := 0; cpu < runtime.NumCPU(); cpu++ {
+		go func(offset uint64) {
+			defer wg.Done()
+			for i := offset; i < n; i += uint64(runtime.NumCPU()) {
+				// NOTE: don't bother trying to optimize generateSpendableKey;
+				// profiling shows that ed25519 key generation consumes far
+				// more CPU time than encoding or hashing.
+				keys[i] = generateSpendableKey(seed, i)
```

Comment:
this is not surprising, but perhaps it's worth including the profiling results here, along with a date. It's possible that at some point we switch to a different ed25519 library, or that more crypto-based hardware accelleration becomes available.